command/agent/agent.go

```diff
@@ -447,6 +447,8 @@ func (a *Agent) finalizeClientConfig(c *clientconfig.Config) error {
 		to configure Nomad to work with Consul.`)
 	}
 
+	c.Node.Canonicalize()
```

Comment:
This potentially sets values, so do it once before passing to the client instead of multiple times within the client.